python/nav/portadmin/handlers.py

```diff
@@ -0,0 +1,204 @@
+#
+# Copyright (C) 2011-2015, 2020 UNINETT
+#
+# This file is part of Network Administration Visualized (NAV).
+#
+# NAV is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.  You should have received a copy of the GNU General Public License
+# along with NAV. If not, see <http://www.gnu.org/licenses/>.
+#
+"""Interface definition for PortAdmin management handlers"""
+from django.utils.encoding import python_2_unicode_compatible
+
+
+@python_2_unicode_compatible
+class ManagementHandler:
+    """Defines a common interface for all types of PortAdmin management handlers.
+
+    This defines the set of methods that a handler class may be expected by PortAdmin
+    to provide, regardless of the underlying management protocol implemented by such
+    a class.
+    """
+    def __init__(self, netbox, **kwargs):
+        self.netbox = netbox
+
+    def __str__(self):
+        # TODO: This was copied from SNMPHandler and doesn't make much sense
+        return self.netbox.type.vendor.id
```

Comment:
Make this one also raise `NotmplementedError`. Might be possible to figure out a generic `__str__` as soon as you have another non-abstract handler in addition to `SNMPHandler`.

Reply:
I'm having a hard time figuring out what this `__str__()` was good for in the `SNMPHandler` class it came from, it just looks silly. Maybe some git archaeology will give answers. I suspect it is useless (and a good `__repr__()` is probably more useful).